quipucords/tests/integration/test_network_scan.py

```diff
@@ -225,6 +228,48 @@ def fingerprint_fact_map():
     }
 
 
+@pytest.fixture
+def unexpected_fingerprints():
+    """Set of facts that wouldn't appear on network scans."""
+    return {
+        # sattelite / vcenter exclusive
+        "virtual_host_name",
+        "virtual_host_uuid",
+        # vcenter exclusive
+        "vm_cluster",
+        "vm_datacenter",
+        "vm_dns_name",
+        "vm_host_core_count",
+        "vm_host_socket_count",
+        "vm_state",
+        "vm_uuid",
+    }
+
+
+def test_sanity_check_raw_fact_matches(
+    expected_network_scan_facts, fingerprint_fact_map, unexpected_fingerprints
+):
+    """Ensure raw facts mapped to fingerprint facts match known facts."""
+    raw_facts_for_fingerprints = reduce(
+        or_operator,
+        (fact_expander(fact) for fact in fingerprint_fact_map.values()),
+    )
```

Comment:
Let's evaluate this together.
This might have a better readability considering the reader wouldn't need to know `reduce` and `or_` (aliased as `or_operator`

```suggestion
    raw_facts_for_fingerprints = set()
    for fact in fingerprint_fact_map.values():
        raw_facts_for_fingerprints |= fact_expander(fact)
```

Reply:
In both cases it is required to know what the or operator does on a set but I'm now more inclined to think Guido was right XD

Reply:
I do prefer the one without reduce()!